### PR TITLE
Refactor word wrapping and on-node properties

### PR DESCRIPTION
**Pull request by alex-mashin**
Refactor word wrapping and on-node properties, to facilitate further changes.

### CODE REVIEW

**Comment by alex-mashin** [comment]
@YOUR1, please take part in the review.

File: tests/phpunit/Unit/Graph/GraphFormatterTest.php [hunk 1]
```diff
@@ -118,7 +119,7 @@ class GraphFormatterTest extends \PHPUnit\Framework\TestCase {
 	 * @return GraphFormatter
 	 */
 	private static function graph( array $case ): GraphFormatter {
-		$graph = new GraphFormatter( new GraphOptions( GraphFormatterTest::BASE_PARAMS + $case['params'] ) );
+		$graph = new GraphFormatter( new GraphOptions( array_merge( GraphFormatterTest::BASE_PARAMS, $case['params'] ) ) );
```
Comment (YOUR1) [inline tests/phpunit/Unit/Graph/GraphFormatterTest.php:122]: Is there any reason why you changed this?

File: src/Graph/GraphFormatter.php [hunk 2]
```diff
@@ -98,54 +98,22 @@ public function buildGraph( $nodes ) {
...
-									. $field['name'] . '</td>'
-									. '<td align="' . $alignment . '">'
-										. $instance->getWordWrappedText(
-											$field['value'],
-											$instance->options->getWordWrapLimit()
-										)
-									. '</td></tr>';
-							}, $fields ) ) . "\n</table>\n>";
+				$label = $nodeLabel ?: $node->getID();
+				$nodeTooltip = htmlspecialchars( $label );
```
Comment (YOUR1) [inline src/Graph/GraphFormatter.php:109]: The previous version replaced the '<br />' with '' when the Diagrams extension was loaded if im correct. What happened to that?

File: tests/phpunit/Unit/Graph/GraphFormatterTest.php (hunk 1 above)
Comment (alex-mashin) [reply tests/phpunit/Unit/Graph/GraphFormatterTest.php:122]: `+` was an error. In this place `$case['params']` should override `GraphFormatterTest::BASE_PARAMS`.

File: src/Graph/GraphFormatter.php (hunk 2 above)
Comment (alex-mashin) [reply src/Graph/GraphFormatter.php:109]: Please elaborate. Replaced where?

Comment (YOUR1) [reply src/Graph/GraphFormatter.php]: Sorry for the confusion; Github did a weird thing with my `<br />`. 

You refactored this part;
```php
if ( version_compare( $wgVersion, '1.33', '>=' ) &&
   \ExtensionRegistry::getInstance()->isLoaded( 'Diagrams' ) ) {
   $nodeTooltip = str_replace( '<br />', '', $nodeTooltip );
}
```

But I can't find that anywhere now.

Comment (alex-mashin) [reply src/Graph/GraphFormatter.php]: So, node tooltip for table nodes (with fields) for Diagrams should have line breaks instead of `<br />`?

Comment (alex-mashin) [reply src/Graph/GraphFormatter.php]: Also, a question: could you please check how Diagrams will treat `\n` escape sequence in a quoted node label or identifier, like `[label="First line\nsecond line"]` or `"First line\second line" -> 'Other node"`?

**Comment by krabina** [comment]
Can we merge this, @YOUR1 ?

**Comment by alex-mashin** [comment]
> Can we merge this, @YOUR1 ?

No, we shouldn't. I am abandoning this request. I have fully rewritten the whole format, only unit tests are remaining.

**Comment by alex-mashin** [comment]
Abandoning in favour of a more significant change.

**Pull request closed without merging**